sympy/sets/sets.py

```diff
@@ -932,6 +944,19 @@ def right_open(self):
         """
         return self._args[3]
 
+    @property
+    def is_empty(self):
+        if self.start == S.NegativeInfinity or self.end == S.Infinity:
+            return False
```

Comment:
Is the above check necessary? I don't like it because it implicitly assumes that we couldn't have an unevaluated closed interval like `Interval(-oo, -oo, left_open=False, right_open=False, evaluate=False)` (That's not currently possible but it *should* be)

Reply:
It has an impact on `Interval(x, oo).is_empty` (for general `x`).

Since currently Intervals are reals-only and SymPy should implicitly assume necessary conditions to hold (https://github.com/sympy/sympy/issues/17488#issuecomment-524037140), I made it evaluate unconditionally to `False` even if `x` hasn't specifically been declared real.

Without the mentioned checks `.is_empty` would return `None`.

Maybe that's for the best, i.e. remove the check and the corresponding failing test. In some contexts it may also be surprising to have `Interval(x, oo).is_empty` automatically evaluate to `False` even if `x == oo` is still a possibility.
I'll change it.

(Personally I'd also find it more elegant if the `is_extended_*` assumptions had a set-theoretic equivalent, so I'm in favor of supporting intervals of extended reals. But I won't claim to understand all of the implications it could have on other parts of the codebase.)

Reply:
How about this:
```
def is_empty(self):
    if self.left_open or self.right_open:
        cond = self.start < self.end  # One/both bounds open
    else:
        cond = self.start <= self.end  # Both bounds closed
    return fuzzy_bool(cond)
```

Reply:
Obviously I got the inequalities the wrong way round :)

Reply:
Somehow my answer by e-mail doesn't show up here. Anyway thanks, marking as resolved.